extensions/interactions/InteractiveMap/directives/oppia-interactive-interactive-map.component.ts

```diff
@@ -29,18 +29,44 @@ import { CurrentInteractionService } from 'pages/exploration-player-page/service
 import { PlayerPositionService } from 'pages/exploration-player-page/services/player-position.service';
 import { Subscription } from 'rxjs';
 import { InteractiveMapRulesService } from './interactive-map-rules.service';
-import { icon, LatLng, latLng, LeafletMouseEvent, marker, TileLayer, tileLayer } from 'leaflet';
+import { icon, LatLng, latLng, LeafletMouseEvent, Marker, marker, TileLayer, tileLayer } from 'leaflet';
 import { downgradeComponent } from '@angular/upgrade/static';
 
+interface OverlayStyle {
+  'background-color'?: string;
+  opacity?: number;
+  'z-index'?: number;
+}
+
+interface MapOptions {
+  center: LatLng;
+  layers: TileLayer[];
+  zoom: number;
+}
+
 @Component({
   selector: 'oppia-interactive-interactive-map',
   templateUrl: './interactive-map-interaction.component.html'
 })
 export class InteractiveInteractiveMapComponent implements OnInit, OnDestroy {
-  @Input() lastAnswer;
-  @Input() latitudeWithValue: string;
-  @Input() longitudeWithValue: string;
-  @Input() zoomWithValue: string;
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  @Input() latitudeWithValue!: string;
+  // Last amswer property only exist in exploration_player view (including the
+  // preview mode). Otherwise, it is null.
+  @Input() lastAnswer!: number[] | null;
+  @Input() longitudeWithValue!: string;
+  @Input() zoomWithValue!: string;
+  // Class Marker is of type Marker<P=any> therefore type
+  // cannot be determined.
```

Comment:
I don't understand this comment. Can you explain or rephrase?

Reply:
this comment is already exist in 2-3 places already for mapMarkers. It means Marker can be of type <any>, but we are unable to use any. So we are using Marker<unknown>.

Reply:
Hmm, I see why we cannot use `Marker<any>` but I'm not sure why we cannot use precise type for the marker, like `Marker<string>` for example

Reply:
Done